core/src/main/java/io/grpc/internal/ServiceConfigUtil.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2018, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.internal;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Verify;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Random;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+/**
+ * Helper utility to work with service configs.
+ */
+final class ServiceConfigUtil {
+
+  private static final Logger logger = Logger.getLogger(ServiceConfigUtil.class.getName());
+
+  // From https://github.com/grpc/proposal/blob/master/A2-service-configs-in-dns.md
+  static final String SERVICE_CONFIG_PREFIX = "_grpc_config=";
```

Comment:
in https://github.com/grpc/proposal/blob/master/A2-service-configs-in-dns.md there is
either "_grpc_config." or "grpc_config=" 
but "_grpc_config="